docs/source/developer_guide/metadata.md

```diff
@@ -148,5 +158,69 @@ tags = runtime_configuration.metadata['tags']
+
+#### Registration
+Schemaspaces are discovered via the entrypoints functionality.  When the `SchemaManager` starts, it fetches all entrypoint instances named by the group `'metadata.schemaspaces'`, and uses the entrypoint data to load each found instance.
+Entrypoints are registered in the package's setup configuration.  
+
+Here is an example of the schemaspace entrypoints defined in Elyra:
+```
+entry_points={
+  'metadata.schemaspaces': [
+    'runtimes = elyra.metadata.schemaspaces:Runtimes',
+    'runtimes-images = elyra.metadata.schemaspaces:RuntimeImages',
+    'code-snippets = elyra.metadata.schemaspaces:CodeSnippets',
+    'component-registries = elyra.metadata.schemaspaces:ComponentRegistries'
+  ]
+}
+```
+As you can see, each entry has a _name_ that equates to a module and object/class.
+
+### SchemasProvider
+As mentioned above, `Schemaspace` instances do not know their set of schemas at the time of their construction.  Instead, the `SchemaManager` loads a different entrypoints group following the loading of schemaspaces named `'metadata.schemas_providers'`.  These entrypoints identify a `SchemasProvider` object and this object is responsible for returning schemas when called upon by the `SchemaManager`.
+`SchemasProvider` instances are concrete instances of the abstract base class `SchemasProvider`.  The class is really quite simple:
+```python
+class SchemasProvider(ABC):
+    """Abstract base class used to obtain schema definitions from registered schema providers."""
+
+    @abstractmethod
+    def get_schemas(self) -> List[Dict]:
+        """Returns a list of schemas"""
+        pass
+```
+When called, the `SchemasProvider` instance is responsible for producing a list of schemas.  These schemas must conform to the metadata service's _meta-schema_ and, as a result, will include the schemaspace name and schemaspace id to which they correspond, in addition to a `name` property representing that schema's name.  The `SchemaManager` uses this meta-information to confirm the schemaspace has been loaded and register that schema with the schemaspace following its successful validation against the meta-schema.
+
+Some SchemasProviders may have schema information which is dynamic, based on the active configuration of the system.  In such cases, they are responsible for returning information pertinent to the current configuration.  For example, the _component-registry_ schema (of the _component-registries_ schemaspace) includes an enumerated property named `runtime`.  However, the values/choices included in the `enum` are a function of what packages are configured.  As a result, the `get_schemas()` method for the _component-registries_ `SchemasProvider` determines, based on installed packages, which entries to add to the `runtime` enumeration.
```

Comment:
Missing markdown

```suggestion
Some `SchemasProvider`s may have schema information which is dynamic, based on the active configuration of the system.  In such cases, they are responsible for returning information pertinent to the current configuration.  For example, the _component-registry_ schema (of the _component-registries_ schemaspace) includes an enumerated property named `runtime`.  However, the values/choices included in the `enum` are a function of what packages are configured.  As a result, the `get_schemas()` method for the _component-registries_ `SchemasProvider` determines, based on installed packages, which entries to add to the `runtime` enumeration.
```

Reply:
Rather than have the 's' in another font, I think using "`SchemasProvider` instances" may render better.